framework/src/postprocessors/NodalVariableValue.C

```diff
@@ -33,15 +33,20 @@ NodalVariableValue::NodalVariableValue(const InputParameters & parameters)
   : GeneralPostprocessor(parameters),
     _mesh(_subproblem.mesh()),
     _var_name(parameters.get<VariableName>("variable")),
-    _node_ptr(_mesh.getMesh().query_node_ptr(getParam<unsigned int>("nodeid"))),
+    _node_ptr(nullptr),
     _scale_factor(getParam<Real>("scale_factor"))
 {
   // This class may be too dangerous to use if renumbering is enabled,
   // as the nodeid parameter obviously depends on a particular
   // numbering.
   if (_mesh.getMesh().allow_renumbering())
     mooseError("NodalVariableValue should only be used when node renumbering is disabled.");
+}
 
+void
+NodalVariableValue::initialSetup()
+{
+  _node_ptr = _mesh.getMesh().query_node_ptr(getParam<unsigned int>("nodeid"));
```

Comment:
What's the change in this PR that required querying the nodeid at the `initialSetup` step?

Reply:
before this PR, there were no instances where this object was used in a truly distributed mesh. This PR triggered mesh distribution, e.g. remote element deletion, for at least one simulation in which this object was present. Late mesh distribution (e.g. late mesh element deletion) occurs after user object construction. So the result of `query_elem_ptr` if run at construction time would be a misnomer if later remote elements (and their connected nodes) are deleted